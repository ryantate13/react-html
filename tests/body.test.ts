import {body} from '../src';

describe('body', () => {
    it('renders with children', () => {
        expect(body({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(body({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(body()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(body({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(body({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(body({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(body({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(body({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(body({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(body({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(body({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(body({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(body({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(body({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(body({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(body({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(body({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(body({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(body({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(body({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(body({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(body({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(body({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(body({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(body({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(body({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(body({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(body({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(body({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(body({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(body({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(body({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(body({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(body({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(body({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(body({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(body({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(body({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(body({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(body({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(body({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(body({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(body({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(body({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(body({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(body({translate: "no"})).toMatchSnapshot();
    });
});