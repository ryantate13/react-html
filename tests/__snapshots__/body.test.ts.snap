// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body renders arrays of children 1`] = `
<body>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</body>
`;

exports[`body renders with accessKey="a,b,c,d" 1`] = `
<body
  accessKey="a,b,c,d"
/>
`;

exports[`body renders with autoCapitalize="characters" 1`] = `
<body
  autoCapitalize="characters"
/>
`;

exports[`body renders with autoCapitalize="none" 1`] = `
<body
  autoCapitalize="none"
/>
`;

exports[`body renders with autoCapitalize="off" 1`] = `
<body
  autoCapitalize="off"
/>
`;

exports[`body renders with autoCapitalize="on" 1`] = `
<body
  autoCapitalize="on"
/>
`;

exports[`body renders with autoCapitalize="sentences" 1`] = `
<body
  autoCapitalize="sentences"
/>
`;

exports[`body renders with autoCapitalize="words" 1`] = `
<body
  autoCapitalize="words"
/>
`;

exports[`body renders with children 1`] = `
<body>
  TEST
</body>
`;

exports[`body renders with className="foo bar baz" 1`] = `
<body
  className="foo bar baz"
/>
`;

exports[`body renders with contentEditable="false" 1`] = `
<body
  contentEditable="false"
/>
`;

exports[`body renders with contentEditable="true" 1`] = `
<body
  contentEditable="true"
/>
`;

exports[`body renders with dir="auto" 1`] = `
<body
  dir="auto"
/>
`;

exports[`body renders with dir="ltr" 1`] = `
<body
  dir="ltr"
/>
`;

exports[`body renders with dir="rtl" 1`] = `
<body
  dir="rtl"
/>
`;

exports[`body renders with draggable="false" 1`] = `
<body
  draggable="false"
/>
`;

exports[`body renders with draggable="true" 1`] = `
<body
  draggable="true"
/>
`;

exports[`body renders with empty attributes 1`] = `<body />`;

exports[`body renders with hidden=false 1`] = `
<body
  hidden={false}
/>
`;

exports[`body renders with hidden=true 1`] = `
<body
  hidden={true}
/>
`;

exports[`body renders with id="TEST" 1`] = `
<body
  id="TEST"
/>
`;

exports[`body renders with inputMode="decimal" 1`] = `
<body
  inputMode="decimal"
/>
`;

exports[`body renders with inputMode="email" 1`] = `
<body
  inputMode="email"
/>
`;

exports[`body renders with inputMode="none" 1`] = `
<body
  inputMode="none"
/>
`;

exports[`body renders with inputMode="numeric" 1`] = `
<body
  inputMode="numeric"
/>
`;

exports[`body renders with inputMode="search" 1`] = `
<body
  inputMode="search"
/>
`;

exports[`body renders with inputMode="tel" 1`] = `
<body
  inputMode="tel"
/>
`;

exports[`body renders with inputMode="text" 1`] = `
<body
  inputMode="text"
/>
`;

exports[`body renders with inputMode="url" 1`] = `
<body
  inputMode="url"
/>
`;

exports[`body renders with is="foo" 1`] = `
<body
  is="foo"
/>
`;

exports[`body renders with itemID="https://google.com/" 1`] = `
<body
  itemID="https://google.com/"
/>
`;

exports[`body renders with itemProp="https://google.com/" 1`] = `
<body
  itemProp="https://google.com/"
/>
`;

exports[`body renders with itemRef="foo bar" 1`] = `
<body
  itemRef="foo bar"
/>
`;

exports[`body renders with itemScope=false 1`] = `
<body
  itemScope={false}
/>
`;

exports[`body renders with itemScope=true 1`] = `
<body
  itemScope={true}
/>
`;

exports[`body renders with itemType="https://google.com/" 1`] = `
<body
  itemType="https://google.com/"
/>
`;

exports[`body renders with lang="" 1`] = `
<body
  lang=""
/>
`;

exports[`body renders with lang="es-419" 1`] = `
<body
  lang="es-419"
/>
`;

exports[`body renders with slot="TEST" 1`] = `
<body
  slot="TEST"
/>
`;

exports[`body renders with spellCheck="false" 1`] = `
<body
  spellCheck="false"
/>
`;

exports[`body renders with spellCheck="true" 1`] = `
<body
  spellCheck="true"
/>
`;

exports[`body renders with style={"color":"#000000"} 1`] = `
<body
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`body renders with tabIndex=1 1`] = `
<body
  tabIndex={1}
/>
`;

exports[`body renders with title="TEST" 1`] = `
<body
  title="TEST"
/>
`;

exports[`body renders with translate="no" 1`] = `
<body
  translate="no"
/>
`;

exports[`body renders with translate="yes" 1`] = `
<body
  translate="yes"
/>
`;

exports[`body renders without attributes 1`] = `<body />`;
